crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Common.hs

```diff
@@ -233,58 +174,58 @@ build_llvm_override fnm args ret args' ret' llvmOverride =
 
-register_1arg_polymorphic_override :: forall p sym arch wptr l a rtp.
+register_1arg_polymorphic_override :: forall p sym ext arch wptr.
   (IsSymInterface sym, HasLLVMAnn sym, HasPtrWidth wptr) =>
   String ->
-  (forall w. (1 <= w) => NatRepr w -> SomeLLVMOverride p sym LLVM) ->
-  RegOverrideM p sym arch rtp l a ()
+  (forall w. (1 <= w) => NatRepr w -> SomeLLVMOverride p sym ext) ->
+  MakeOverride p sym ext arch
 register_1arg_polymorphic_override prefix overrideFn =
-  do (L.Declare{ L.decName = L.Symbol nm },_,_) <- ask
-     case List.stripPrefix prefix nm of
-       Just ('.':'i': (readDec -> (sz,[]):_))
-         | Some w <- mkNatRepr sz
-         , Just LeqProof <- isPosNat w
-         -> case overrideFn w of SomeLLVMOverride ovr -> register_llvm_override ovr
-       _ -> empty
-
-basic_llvm_override :: forall p args ret sym arch wptr l a rtp.
+  MakeOverride $ \(L.Declare{ L.decName = L.Symbol nm }) _ _ ->
+    case List.stripPrefix prefix nm of
+      Just ('.':'i': (readDec -> (sz,[]):_))
+        | Some w <- mkNatRepr sz
+        , Just LeqProof <- isPosNat w
+        -> Just (overrideFn w)
+      _ -> Nothing
+
+basic_llvm_override :: forall p args ret sym ext arch wptr.
   (IsSymInterface sym, HasLLVMAnn sym, HasPtrWidth wptr) =>
-  LLVMOverride p sym LLVM args ret ->
-  OverrideTemplate p sym arch rtp l a
+  LLVMOverride p sym ext args ret ->
+  OverrideTemplate p sym ext arch
 basic_llvm_override ovr = OverrideTemplate matcher regOvr
   where
     ovrDecl = llvmOverride_declare ovr
     L.Symbol ovrNm = L.decName ovrDecl
     isDarwin = Info.os == "darwin"
 
-    matcher :: TemplateMatcher
-    matcher | isDarwin  = DarwinAliasMatch ovrNm
-            | otherwise = ExactMatch ovrNm
+    matcher :: Match.TemplateMatcher
+    matcher | isDarwin  = Match.DarwinAliasMatch ovrNm
+            | otherwise = Match.ExactMatch ovrNm
 
-    regOvr :: RegOverrideM p sym arch rtp l a ()
+    regOvr :: MakeOverride p sym ext arch
     regOvr = do
-      (requestedDecl, _ ,_) <- ask
-      let L.Symbol requestedNm = L.decName requestedDecl
-      -- If we are on Darwin and the function name contains Darwin-specific
-      -- prefixes or suffixes, change the name of the override to the name
-      -- containing prefixes/suffixes. See Note [Darwin aliases] for an
-      -- explanation of why we do this.
-      let ovr' | isDarwin
-               , ovrNm == stripDarwinAliases requestedNm
-               = ovr { llvmOverride_declare =
-                         ovrDecl { L.decName = L.Symbol requestedNm }}
-
-               | otherwise
-               = ovr
-      register_llvm_override ovr'
+      MakeOverride $ \requestedDecl _ _ -> do
+        let L.Symbol requestedNm = L.decName requestedDecl
+        -- If we are on Darwin and the function name contains Darwin-specific
+        -- prefixes or suffixes, change the name of the override to the name
+        -- containing prefixes/suffixes. See Note [Darwin aliases] for an
```

Comment:
`Note [Darwin aliases]` now lives in a different module, so make sure to say `See Note [Darwin aliases] in Lang.Crucible.LLVM.Intrinsics.Match [...]` to clarify.